lib_eio/unix/private.ml

```diff
@@ -7,7 +7,7 @@ type _ Effect.t +=
   | Await_readable : Unix.file_descr -> unit Effect.t
   | Await_writable : Unix.file_descr -> unit Effect.t
   | Get_monotonic_clock : Eio.Time.Mono.ty r Effect.t
-  | Pipe : Switch.t -> (source_ty r * sink_ty r) Effect.t
+  | Pipe : Switch.t -> ([< source_ty] r * [< sink_ty] r) Effect.t
```

Comment:
You shouldn't need this. Just cast the result.

Reply:
If I'm using 
```
let pipe (type tag) ~sw ((Resource.T (v, ops)) : [> tag mgr_ty] r) =
  let module X = (val (Resource.get ops Pi.Mgr)) in
  let r, w = X.pipe v ~sw in
  let r = (r : [Flow.source_ty | Resource.close_ty] r :> [< Flow.source_ty | Resource.close_ty] r) in
  let w = (w : [Flow.sink_ty   | Resource.close_ty] r :> [< Flow.sink_ty   | Resource.close_ty] r) in
  r, w
``` , then i dont think I'll have to cast the result in a pipe. Right?